algebra/tests/matrix/various.c

```diff
@@ -483,3 +483,136 @@ TEST_GROUP_RUNNER(group_matrix_times)
 	RUN_TEST_CASE(group_matrix_times, matrix_times_nan);
 	RUN_TEST_CASE(group_matrix_times, matrix_times_nanTrp);
 }
+
+
+/* ##############################################################################
+ * -----------------        matrix_writeSubmatrix tests       -------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_matrix_writeSubmatrix);
+
+
+TEST_SETUP(group_matrix_writeSubmatrix)
+{
+	/* M1 = F */
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK,
+		algebraTests_createAndFill(&M1, buffs_rowsF, buffs_colsF, buffs_F, buffs_colsF * buffs_rowsF));
+
+	/* M2 = B */
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK,
+		algebraTests_createAndFill(&M2, buffs_rowsB, buffs_colsB, buffs_B, buffs_colsB * buffs_rowsB));
+
+	/* M3 = B */
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, algebraTests_matrixCopy(&M3, &M1));
+}
+
+
+TEST_TEAR_DOWN(group_matrix_writeSubmatrix)
+{
+	matrix_bufFree(&M1);
+	matrix_bufFree(&M2);
+	matrix_bufFree(&M3);
+}
+
+
+TEST(group_matrix_writeSubmatrix, matrix_writeSubmatrix_onStart)
+{
+	int rowToWrite = 0, colToWrite = 0;
+
+	TEST_ASSERT_EQUAL_INT(WRITE_SUBMAT_OK, matrix_writeSubmatrix(&M1, rowToWrite, colToWrite, &M2));
+
+	TEST_ASSERT_EQUAL_INT(CHECK_OK, algebraTests_writeSubmatCheck(&M3, rowToWrite, colToWrite, &M2, &M1));
+}
+
+
+TEST(group_matrix_writeSubmatrix, matrix_writeSubmatrix_inMiddle)
+{
+	int rowToWrite = 5, colToWrite = 2;
```

Comment:
Magic numbers. How about:
```suggestion
	unsigned row, col;
	
	row = matrix_rowsGet(M1) - matrix_rowsGet(M2);
	col = matrix_colsGet(M1) - matrix_colsGet(M2);
```
so that you write it at the end of the matrix (botton right corner)?

Also: we can stick to pure `col`, `row` notation